interpreter/interpreter_test.go

```diff
@@ -515,6 +515,25 @@ var (
 				"input": "kathmandu",
 			},
 		},
+		{
+			name:  "nested_proto_field",
+			expr:  `pb3.single_nested_message.bb`,
+			types: []proto.Message{&proto3pb.TestAllTypes{}},
+			env: []*exprpb.Decl{
+				decls.NewVar("pb3",
+					decls.NewObjectType("google.expr.proto3.test.TestAllTypes")),
+			},
+			in: map[string]interface{}{
+				"pb3": &proto3pb.TestAllTypes{
+					NestedType: &proto3pb.TestAllTypes_SingleNestedMessage{
+						SingleNestedMessage: &proto3pb.TestAllTypes_NestedMessage{
+							Bb: 1234,
+						},
+					},
+				},
+			},
+			out: types.Int(1234),
+		},
```

Comment:
So as I understand the comment, this test would pass without these changes. Could you have a test that fails without the changes? Also for an expression that uses indexing.

Reply:
The code is remarkably robust in that I only noticed the issue when looking at alloc counts from benchmarks. I'll see if I can add a test, but I think the nice thing about the current arrangement is that almost no matter what I do, the code will work. :)